examples/cvode/serial/cvRocket_dns.c

```diff
@@ -0,0 +1,417 @@
+#define Y2    RCONST(0.0)
+#define RTOL  RCONST(1.0e-5)  /* scalar relative tolerance            */
+#define ATOL1 RCONST(1.0e-2)  /* vector absolute tolerance components */
+#define ATOL2 RCONST(1.0e-1)
+#define T0    RCONST(0.0)     /* initial time           */
+#define T1    RCONST(1.0)     /* first output time      */
+#define TINC  RCONST(1.0)     /* output time increment  */
+#define NOUT  70              /* number of output times */
+
+#define ZERO RCONST(0.0)
+
+/* Functions Called by the Solver */
+
+static int f(sunrealtype t, N_Vector y, N_Vector ydot, void* user_data);
+
+static int g(sunrealtype t, N_Vector y, sunrealtype* gout, void* user_data);
+
+static int Jac(sunrealtype t, N_Vector y, N_Vector fy, SUNMatrix J, void* user_data, N_Vector tmp1, N_Vector tmp2,
+               N_Vector tmp3);
+
+/* Private functions to output results */
+static void PrintOutput(sunrealtype t, sunrealtype y1, sunrealtype y2);
+static void PrintRootInfo(int root_f1, int root_f2, int numroot);
+
+/* Private function to print final statistics */
+static void PrintFinalStats(void* cvode_mem);
+
+/* Private function to check function return values */
+static int check_retval(void* returnvalue, const char* funcname, int opt);
+
+/*
+ *-------------------------------
+ * Main Program
+ *-------------------------------
+ */
+
+int main()
+{
+  SUNContext sunctx;
+  sunrealtype reltol, t, tout;
+  N_Vector y, abstol;
+  SUNMatrix A;
+  SUNLinearSolver LS;
+  void* cvode_mem;
+  int retval, retvalr, iout;
+  int numroot;
+  int rootsfound[2];
+  sunbooleantype engine_on;
+
+  y = abstol = NULL;
+  A          = NULL;
+  LS         = NULL;
+  cvode_mem  = NULL;
+
+  /* Create the SUNDIALS context */
+  retval = SUNContext_Create(NULL, &sunctx);
+  if (check_retval(&retval, "SUNContext_Create", 1))
+    return (1);
+
+  /* Create serial vector of length NEQ for I.C. and abstol */
+  y = N_VNew_Serial(NEQ, sunctx);
+  if (check_retval((void*)y, "N_VNew_Serial", 0))
+    return (1);
+  abstol = N_VNew_Serial(NEQ, sunctx);
+  if (check_retval((void*)abstol, "N_VNew_Serial", 0))
+    return (1);
+
+  /* Initialize y */
+  NV_Ith_S(y, 0) = Y1;
+  NV_Ith_S(y, 1) = Y2;
+
+  /* Set the scalar relative tolerance */
+  reltol = RTOL;
+  /* Set the vector absolute tolerance */
+  NV_Ith_S(abstol, 0) = ATOL1;
+  NV_Ith_S(abstol, 1) = ATOL2;
+
+  /* Call CVodeCreate to create the solver memory and specify the Backward Differentiation Formula
+   * */
+  cvode_mem = CVodeCreate(CV_BDF, sunctx);
+  if (check_retval((void*)cvode_mem, "CVodeCreate", 0))
+    return (1);
+
+  /* Call CVodeInit to initialize the integrator memory and specify the right-hand side function in
+   * y'=f(t,y), the inital time T0, and the initial dependent variable vector y. */
+  retval = CVodeInit(cvode_mem, f, T0, y);
+  if (check_retval(&retval, "CVodeInit", 1))
+    return (1);
+
+  /* Call CVodeSVtolerances to specify the scalar relative tolerance and vector absolute tolerances
+   * */
```

Comment:
Why is there a second [empty] comment line?

Reply:
Auto-formatting broke the line because of the length.